### ISSUE TITLE
NeuroKit Warning: ecg_hrv(): Correlation Dimension. Error:  NeuroKit warning: complexity_entropy_multiscale(): Signal might be to short to compute SampEn for scale factors > 0. Setting max_scale_factor to 0.

### ISSUE THREAD

**Issue by Sanjay1995** [neuropsychology/NeuroKit.py#61]
I am giving the data in form of 1D numpy array which is basically i, ii, iii, v1, v2, v3, v4, v5, v6, avr, avl, avf features from PTB database. I call this function nk.bio_process(ecg=ecg_signal[:,i],ecg_quality_model=None). It operates good on several columns but gives error after 2 iterations. Moreover, I have checked data is fine enough in all columns. can anyone please resolve the issue

**Comment by DominiqueMakowski** [comment]
Hi @Sanjay1995, could you provide an example of your dataset? We'll try to fix the error. It might be related to a recent change in HRV computation (#58). Also linking @gattia just in case :)

**Comment by Sanjay1995** [comment]
@DominiqueMakowski

**Comment by gattia** [comment]
Based on the print out it seems that the sample entropy is having problems on the very first pass at the full resolution scale for the multi-scale analysis (this should be the same as just running sample entropy on the full data). This shouldn't be a problem from anything done to the multiscale entropy function recently.  

If I were debugging it, I'd be interested in what the data that is being passed to `complexity_entropy_multiscale()` looks like - whats it's shape, min, max values, how does it looked graphed out, etc.

**Comment by Sanjay1995** [comment]
![screenshot from 2018-04-14 18-23-07](https://user-images.githubusercontent.com/18687010/38768721-7102e66a-4011-11e8-8c86-9e0f5313a16b.png)

**Comment by Sanjay1995** [comment]
this is how my one data column which I (i) as I mentioned above looks like

**Comment by Sanjay1995** [comment]
@gattia

**Comment by Sanjay1995** [comment]
thanks @DominiqueMakowski it is really useful, but tell me ecg_preprocess() and bio_process() works alike in my case?

**Comment by DominiqueMakowski** [comment]
@Sanjay1995  yes bio_process is just a wrapper for processing multiple signals (ECG, EDA, EMG etc.) at once. Using bio-process with only ecg is similar to using `ecg_process`. However, `ecg_process` uses itself, the `ecg_preprocess` function that only does low level preprocessing (mainly extracting R peaks and not computing more complex indices such as HRV for example).

**Comment by Sanjay1995** [comment]
as I have already mentioned you that I am using dataset of PTB ecg dataset, and your library also inculcates it. But ecg_preprocess() also fails on some signals giving error (index 0 is out of bounds for axis 0) don't know why.

**Comment by Sanjay1995** [comment]
If I only process LEAD 1 (i in my data), then would it give features which ('T_Waves', 'Cardiac_Cycles', 'P_Waves', 'Q_Waves', 'HRV', 'R_Peaks') helps me in classification of heart disease class.

**Comment by DominiqueMakowski** [comment]
yes, you should use only the column of the dataset corresponding to `i`. I used the full PTB dataset only to create a machine learning model that automatically classifies the provided lead signal and returns the probability of correct classification (a proxy of [signal quality](http://neurokit.readthedocs.io/en/latest/features/ecg_quality.html)). But for investigating ECG features using only LEAD I is sufficient.

**Comment by Sanjay1995** [comment]
@DominiqueMakowski  thanks.

**Comment by waleedkaimkhani** [comment]
@DominiqueMakowski I am getting same error ecg_process index 0 is out of bound error but I have the signal of length greater than 1

**Comment by DominiqueMakowski** [comment]
@waleedkaimkhani could you provide a sample of your data? thanks

**Comment by waleedkaimkhani** [comment]
my dataset is ptb ecg database. you can check it on https://www.physionet.org/physiobank/database/ptbdb/

**Comment by DominiqueMakowski** [comment]
haha alright;

1) did you select correctly a **one dimensional** array OR **one** pandas' dataframe column (corresponding to LEAD 1)? 
2) if yes, could you save (in txt, csv or json) this unique column or array and attach it here so I can check directly with the exact input you provide to neurokit's routines? Thanks 😅

**Comment by waleedkaimkhani** [comment]
![screenshot from 2018-04-14 22 35 00](https://user-images.githubusercontent.com/31384546/38770972-42dcfe60-4034-11e8-915b-36611117b2c3.png)

**Comment by DominiqueMakowski** [comment]
or send it to me dom.makowski@gmail.com

**Comment by waleedkaimkhani** [comment]
@DominiqueMakowski   i have sent you mail

**Comment by DominiqueMakowski** [comment]
@waleedkaimkhani your code should look like that

```python
import neurokit as nk
import pandas as pd

df = pd.read_csv("file.csv")
ecg_processed = nk.ecg_process(ecg=df["i"], sampling_rate=1000)
```
